diff --git a/lib/add-channel.js b/lib/add-channel.js
index 7dfdaab6e2b8c517e8d706b33461a2769f0783a5..ae70cdc010755d50500d21f10cd2043a215e4249 100644
--- a/lib/add-channel.js
+++ b/lib/add-channel.js
@@ -19,7 +19,7 @@ export default async function (npmrc, { npmPublish }, pkg, context) {
 
     logger.log(`Adding version ${version} to npm registry on dist-tag ${distTag}`);
     const result = execa(
-      "npm",
+      "pnpm",
       ["dist-tag", "add", `${pkg.name}@${version}`, distTag, "--userconfig", npmrc, "--registry", registry],
       {
         cwd,
diff --git a/lib/get-registry.js b/lib/get-registry.js
index 4cb42dea3f0072a18ceb297a5bf90408743ea47c..c4f9e1ebfd5dbfc5903345fc0c25e1bd659123b9 100644
--- a/lib/get-registry.js
+++ b/lib/get-registry.js
@@ -9,7 +9,7 @@ export default function ({ publishConfig: { registry } = {}, name }, { cwd, env
     getRegistryUrl(
       name.split("/")[0],
       rc(
-        "npm",
+        "pnpm",
         { registry: "https://registry.npmjs.org/" },
         { config: env.NPM_CONFIG_USERCONFIG || path.resolve(cwd, ".npmrc") }
       )
diff --git a/lib/prepare.js b/lib/prepare.js
index 3e76bec44cf595a1b4141728336bed904d4d518d..2a1d71aca608230122af574322676716343d66c2 100644
--- a/lib/prepare.js
+++ b/lib/prepare.js
@@ -10,9 +10,10 @@ export default async function (
   const basePath = pkgRoot ? path.resolve(cwd, pkgRoot) : cwd;
 
   logger.log("Write version %s to package.json in %s", version, basePath);
+  logger.log('Packing bundle to %s', basePath)
 
   const versionResult = execa(
-    "npm",
+    "pnpm",
     ["version", version, "--userconfig", npmrc, "--no-git-tag-version", "--allow-same-version"],
     {
       cwd: basePath,
@@ -27,7 +28,7 @@ export default async function (
 
   if (tarballDir) {
     logger.log("Creating npm package version %s", version);
-    const packResult = execa("npm", ["pack", basePath, "--userconfig", npmrc], { cwd, env, preferLocal: true });
+    const packResult = execa("pnpm", ["pack"], { cwd: basePath, env, preferLocal: true });
     packResult.stdout.pipe(stdout, { end: false });
     packResult.stderr.pipe(stderr, { end: false });
 
diff --git a/lib/publish.js b/lib/publish.js
index ac3e6feea8e388d88e77d58538a265f7ab4eac05..dd5653348713eeb26e89681421b6524e27f83623 100644
--- a/lib/publish.js
+++ b/lib/publish.js
@@ -21,7 +21,7 @@ export default async function (npmrc, { npmPublish, pkgRoot }, pkg, context) {
 
     logger.log(`Publishing version ${version} to npm registry on dist-tag ${distTag}`);
     const result = execa(
-      "npm",
+      "pnpm",
       ["publish", basePath, "--userconfig", npmrc, "--tag", distTag, "--registry", registry],
       { cwd, env, preferLocal: true }
     );
diff --git a/lib/verify-auth.js b/lib/verify-auth.js
index 99e138e98190cf2545937c9d18e426b3ac4aa53d..b3dda5341b710654bfdd97ca1f18ec231b53fb42 100644
--- a/lib/verify-auth.js
+++ b/lib/verify-auth.js
@@ -18,7 +18,7 @@ export default async function (npmrc, pkg, context) {
 
   if (normalizeUrl(registry) === normalizeUrl(DEFAULT_NPM_REGISTRY)) {
     try {
-      const whoamiResult = execa("npm", ["whoami", "--userconfig", npmrc, "--registry", registry], {
+      const whoamiResult = execa("pnpm", ["whoami", "--userconfig", npmrc, "--registry", registry], {
         cwd,
         env,
         preferLocal: true,